bundles/org.openhab.binding.qbus/src/main/java/org/openhab/binding/qbus/internal/handler/QbusRolHandler.java

```diff
@@ -0,0 +1,357 @@
+        if (QBridgeHandler == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.CONFIGURATION_ERROR,
+                    "No communication with Qbus Bridge!");
+            return;
+        }
+        this.sn = QBridgeHandler.getSn();
+        ;
+    }
+
+    /**
+     * Handle the status update from the thing
+     */
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        QbusCommunication QComm = getCommunication("Screen/Store", rolId);
+
+        if (QComm == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "Bridge communication not initialized when trying to execute command for Screen/Store " + rolId);
+            return;
+        }
+
+        Map<Integer, QbusRol> rolComm = QComm.getRol();
+
+        if (rolComm != null) {
+            QbusRol QRol = rolComm.get(rolId);
+            if (QRol == null) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                        "Bridge communication not initialized when trying to execute command for ROL " + rolId);
+                return;
+            } else {
+                scheduler.submit(() -> {
+                    if (!QComm.communicationActive()) {
+                        restartCommunication(QComm, "Screen/Store", rolId);
+                    }
+
+                    if (QComm.communicationActive()) {
+
+                        if (command == REFRESH) {
+                            handleStateUpdate(QRol);
+                            return;
+                        }
+
+                        switch (channelUID.getId()) {
+                            case CHANNEL_ROLLERSHUTTER:
+                                handleScreenposCommand(QRol, command);
+                                updateStatus(ThingStatus.ONLINE);
+                                break;
+
+                            case CHANNEL_SLATS:
+                                handleSlatsposCommand(QRol, command);
+                                updateStatus(ThingStatus.ONLINE);
+                                break;
+
+                            default:
+                                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                                        "Channel unknown " + channelUID.getId());
+                        }
+                    }
+                });
+            }
+        }
+    }
+
+    /**
+     *
+     * @param message
+     */
+    public void thingOffline(String message) {
+        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR, message);
+    }
+
+    /**
+     * Executes the command for screen up/down position
+     */
+    private void handleScreenposCommand(QbusRol QRol, Command command) {
+        @Nullable
+        String snr = getSN();
+        if (command instanceof org.openhab.core.library.types.UpDownType) {
```

Comment:
You could add the type to the imports. Same for below.
```suggestion
        if (command instanceof UpDownType) {
```